### PR TITLE
fix: various custom operator conformance fixes

### LINKED ISSUE #1774: Update flagd-testbed to v3.6.2 and fix in-process evaluation edge cases
## Summary

Update the flagd-testbed dependency to [v3.6.2](https://github.com/open-feature/flagd-testbed/releases/tag/v3.6.2) and fix the in-process evaluation to pass all new edge-case tests.

Testbed v3.6.2 adds coverage for the edge cases identified in:
- open-feature/flagd#1873 (sem_ver parsing inconsistencies)
- open-feature/flagd#1874 (custom operator error return values)
- open-feature/flagd#1875 ($evaluators/$ref resolution gaps)

### sem_ver (open-feature/flagd#1873)

1. **sem_ver accepts partial versions (`"1.2"`, `"1"`)** by coercing missing components to 0; should reject partial versions to match the standardized behavior (strict X.Y.Z required)

### Custom operator error returns (open-feature/flagd#1874)

2. **`fractional` throws `JsonLogicEvaluationException` on no-bucket-match**; should return `null` to trigger default variant fallback instead of propagating an exception

### $evaluators/$ref resolution (open-feature/flagd#1875)

3. **No diagnostic feedback on unresolved `$ref`s at parse time**; failure only surfaces at evaluation time via `JsonLogicEvaluationException`
4. **Non-deterministic `$evaluators` replacement ordering** (`HashMap` iteration); should use deterministic iteration

Parent: open-feature/flagd#1904

### PR DESCRIPTION
**Pull request by toddbaert**
fix: various custom operator conformance fixes

* support v/V prefix in semver
* support partial versions in semver
* support numeric context values in semver
* return null on errors
* fix fractional single-entry flattening

:warning: wait until flagd/RPC is updated to merge.

Fixes: https://github.com/open-feature/java-sdk-contrib/issues/1774

---

Related:
* https://github.com/open-feature/flagd/pull/1950
* https://github.com/open-feature/js-sdk-contrib/pull/1519
* https://github.com/open-feature/dotnet-sdk-contrib/pull/635
* https://github.com/open-feature/python-sdk-contrib/pull/386
* https://github.com/open-feature/go-sdk-contrib/pull/874

### CODE REVIEW

File: tools/flagd-core/src/main/java/dev/openfeature/contrib/tools/flagd/core/targeting/Fractional.java [hunk 1]
```diff
@@ -36,13 +36,27 @@ public Object evaluate(List arguments, Object data, String jsonPath) throws Json
...
+                List<Object> wrapped = new ArrayList<>();
+                wrapped.add(arg1);
+                for (Object r : remaining) {
+                    wrapped.add(r);
+                }
+                distributions = new ArrayList<>();
+                distributions.add(wrapped);
+            } else {
+                distributions = Arrays.asList(remaining);
+            }
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply tools/flagd-core/src/main/java/dev/openfeature/contrib/tools/flagd/core/targeting/Fractional.java]: Agreed. Fixed. Also made improvements around the `$ref` replace.

**Review by aepfli — APPROVED** [review]

**Review by chrfwow — APPROVED** [review]

File: tools/flagd-core/src/main/java/dev/openfeature/contrib/tools/flagd/core/targeting/Fractional.java [hunk 2]
```diff
@@ -51,7 +61,7 @@ public Object evaluate(List arguments, Object data, String jsonPath) throws Json
             }
```
Comment (leakonvalinka) [inline tools/flagd-core/src/main/java/dev/openfeature/contrib/tools/flagd/core/targeting/Fractional.java]: Do we also need to check for flattening here for single-entry fractionals? Or does that not happen here because the pre-evaluation doesn't "get confused" by 2 different list entry types?

File: tools/flagd-core/src/test/java/dev/openfeature/contrib/tools/flagd/core/targeting/FractionalTest.java [hunk 3]
```diff
@@ -80,4 +82,39 @@ static class TestData {
...
+        Map<String, String> flagdProperties = new HashMap<>();
+        flagdProperties.put(FLAG_KEY, "flagA");
+        data.put(FLAGD_PROPS_KEY, flagdProperties);
+        // no TARGET_KEY set
+
+        List<Object> rule = List.of(
+                // bucket key is a null var result (simulated by being a non-string, non-list)
+                List.of("one", 50), List.of("two", 50));
+
+        // targeting key is null, so fractional falls back to flagKey + targetingKey
```
Comment (leakonvalinka) [inline tools/flagd-core/src/test/java/dev/openfeature/contrib/tools/flagd/core/targeting/FractionalTest.java]: nit-pick, do you mean "bucketing key is null"?

File: tools/flagd-core/src/main/java/dev/openfeature/contrib/tools/flagd/core/targeting/Fractional.java (hunk 2 above)
Comment (toddbaert) [reply tools/flagd-core/src/main/java/dev/openfeature/contrib/tools/flagd/core/targeting/Fractional.java]: Great catch.

File: tools/flagd-core/src/test/java/dev/openfeature/contrib/tools/flagd/core/targeting/FractionalTest.java (hunk 3 above)
Comment (toddbaert) [reply tools/flagd-core/src/test/java/dev/openfeature/contrib/tools/flagd/core/targeting/FractionalTest.java]: Yes!

Comment (toddbaert) [reply tools/flagd-core/src/test/java/dev/openfeature/contrib/tools/flagd/core/targeting/FractionalTest.java]: [Fixed.](https://github.com/open-feature/java-sdk-contrib/pull/1778/changes/378a06ca1a1170f0261c0c2b4c65cc76f393eccf)

File: tools/flagd-core/src/main/java/dev/openfeature/contrib/tools/flagd/core/targeting/Fractional.java (hunk 2 above)
Comment (toddbaert) [reply tools/flagd-core/src/main/java/dev/openfeature/contrib/tools/flagd/core/targeting/Fractional.java]: [Fixed.](https://github.com/open-feature/java-sdk-contrib/pull/1778/changes/378a06ca1a1170f0261c0c2b4c65cc76f393eccf)

File: .gitmodules [hunk 4]
```diff
@@ -4,7 +4,6 @@
 [submodule "providers/flagd/test-harness"]
 	path = providers/flagd/test-harness
 	url = https://github.com/open-feature/test-harness.git
-	branch = v3.5.0
```
Comment (aepfli) [inline .gitmodules:7]: this breaks renovates updates to tags - we should keep this so renovate automagically updates to tags within repos, instead of unreleased hashes
